### ISSUE TITLE
CUDA Illegal Memory Access Error

### ISSUE THREAD

**Issue by chinmay5** [pytorch/pytorch#39935]
## 🐛 Bug

Following is the code snippet I am running 
```
@staticmethod
    def backward(ctx, grad_output):
         grad_label = grad_output.clone()
        num_ft = grad_output.shape[0]
        # grad_label.data.resize_(num_ft, 32, 41)
        lin_indices_3d, lin_indices_2d = ctx.saved_variables
        num_ind = lin_indices_3d.data[0]
        grad_label.data.view(num_ft, -1).index_copy_(1, lin_indices_2d.data[1:1 + num_ind],
                                                     torch.index_select(grad_output.data.contiguous().view(num_ft, -1),
                                                                        1, lin_indices_3d.data[1:1 + num_ind]))
        # raw_input('sdflkj')
        return grad_label, None, None, None
```

During the backward pass, I keep getting the error of CUDA Illegal Memory access.

![image](https://user-images.githubusercontent.com/16525717/84475989-c6915180-ac8d-11ea-82cc-b4efbd89a64d.png)

The code was running as expected with older version of Pytorch (0.4) and I am trying to upgrade it to the newer version but keep getting this strange error.

 - PyTorch Version (e.g., 1.3): (although the error persists on 1.4 and 1,5)
 - OS : Ubuntu 18.04
 - How you installed PyTorch : pip
 - Build command you used (if compiling from source): None
 - Python version: 3.6.8
 - CUDA/cuDNN version: 10.1
 
I guess a few indexing related operations have been changed in recent version but I am not sure why it affects this particular code block. Any help would be highly appreciated

cc @ezyang @gchanan @zou3519 @ngimel

**Comment by chinmay5** [comment]
On running with CUDA_LAUNCH_BLOCKING=1, I get the error
![image](https://user-images.githubusercontent.com/16525717/84478147-1cb3c400-ac91-11ea-8391-3b4cac7e38dc.png)

**Comment by ngimel** [comment]
Please provide a complete runnable example reproducing the error.

**Comment by chinmay5** [comment]
Here is the minimal code to reproduce the error (I got to this point after a bit of debugging please refer https://discuss.pytorch.org/t/loss-backward-throwing-cuda-errors/85221/7)

```
import torch

a = torch.rand([128, 442368, 5], device="cuda", requires_grad=True)

mod = torch.nn.MaxPool1d(5)

out = mod(a)

out.backward(torch.ones_like(out))
```

This is on a new venv with nightly build
![image](https://user-images.githubusercontent.com/16525717/84528232-95da0800-acdf-11ea-8df8-d171998a5eef.png)

Finally the error is

![image](https://user-images.githubusercontent.com/16525717/84528351-d5a0ef80-acdf-11ea-81b8-68787d3ec80f.png)

**Comment by ngimel** [comment]
Can't repro on a fresh build, please make sure that there are no other pytorch installs available in your environment.

**Comment by chinmay5** [comment]
![image](https://user-images.githubusercontent.com/16525717/84529006-fc135a80-ace0-11ea-9a25-69bb0cb38b12.png)

For installation I used the steps as indicated in the screenshot and this is the only version of Pytorch available in the environment

**Comment by ngimel** [comment]
Check that your commands are as recommended on pytorch.org, and check your pytorch version. It does not repro on nightly
```
In [1]: import torch                                                                                                                                                                                  

In [2]: print(torch.__version__)                                                                                                                                                                      
1.6.0.dev20200610+cu101

In [3]: import torch 
   ...:  
   ...: a = torch.rand([128, 442368, 5], device="cuda", requires_grad=True) 
   ...:  
   ...: mod = torch.nn.MaxPool1d(5) 
   ...:  
   ...: out = mod(a) 
   ...:  
   ...: out.backward(torch.ones_like(out))                                                                                                                                                            

```

**Comment by chinmay5** [comment]
Yes, you're right. I was able to run the code snippet. I think the nightly version has the fix. Any idea when would the version 1.6 be out so that I can use a more stable version

**Comment by chinmay5** [comment]
Fixed with nightly build
